src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -1433,6 +1435,37 @@ private async Task WriteStreamPropertiesAsync(SelectExpandNode selectExpandNode,
             }
         }
 
+        private IEnumerable<KeyValuePair<IEdmNavigationProperty, Type>> GetNavigationPropertiesForDeepInsert(SelectExpandNode selectExpandNode, ResourceContext resourceContext)
+        {
+            ISet<IEdmNavigationProperty> navigationProperties = selectExpandNode.SelectedNavigationProperties;
+            object instance = resourceContext.ResourceInstance;
+            PropertyInfo[] properties = instance.GetType().GetProperties();
+            Dictionary<string, object> propertyNamesAndValues = new Dictionary<string, object>();
+
+            foreach (PropertyInfo propertyInfo in properties)
+            {
+                string name = propertyInfo.Name;
+                object value = propertyInfo.GetValue(instance, null);
+                propertyNamesAndValues.Add(name, value);
+            }
+
+            if (navigationProperties != null)
+            {
+                foreach (IEdmNavigationProperty navigationProperty in navigationProperties)
+                {
+                    if (propertyNamesAndValues.ContainsKey(navigationProperty.Name))
```

Comment:
Use `TryGetValue` instead of `ContainsKey`

Reply:
Fixed